crypto/params.c

```diff
@@ -889,8 +889,6 @@ int OSSL_PARAM_set_utf8_ptr(OSSL_PARAM *p, const char *val)
     if (p == NULL)
         return 0;
     p->return_size = 0;
-    if (val == NULL)
-        return 0;
     return set_ptr_internal(p, val, OSSL_PARAM_UTF8_PTR, strlen(val) + 1);
```

Comment:
I wonder, how does `strlen()` treat a NULL pointer?

Reply:
AFAIK, badly. Or even worse, implementation-dependent.

Reply:
Right...  so this needs to be fixed somehow.  Something like this, perhaps?

``` C
    return set_ptr_internal(p, val, OSSL_PARAM_UTF8_PTR, (val == NULL ? 0 : strlen(val) + 1));
```

Reply:
Fixed.